### PR TITLE
Testing utils

### PR DESCRIPTION
**Pull request by mortonjt**
This is a really simple PR to simulate band tables (as shown in the horseshoe paper) and block diagonal matrices.

### CODE REVIEW

File: gneiss/util.py [hunk 1]
```diff
@@ -317,3 +317,74 @@ def random_tree(n):
...
+
+        mat[lower_row:upper_row, lower_col:upper_col] = B
+
+    # Make last block fill in the remainder
+    B = np.random.uniform(size=(nrows-upper_row, ncols-upper_col))
+    mat[upper_row:, upper_col:] = B
+    return mat
+
+
+def _shift(l, n):
```
Comment (wasade) [inline gneiss/util.py]: can an english description of this algorithm be provided? is it just rolling the vector? if so, can `np.roll` be used?

File: gneiss/tests/test_util.py [hunk 2]
```diff
@@ -345,6 +347,37 @@ def test_random_tree(self):
...
+    def test_block_diagonal_4x4(self):
+        np.random.seed(0)
+        res = block_diagonal(4, 4, 2)
+        exp = np.array([[0.5488135, 0.71518937, 0., 0.],
+                        [0.60276338, 0.54488318, 0., 0.],
+                        [0., 0., 0.4236548, 0.64589411],
+                        [0., 0., 0.43758721, 0.891773]])
+        npt.assert_allclose(res, exp, rtol=1e-5, atol=1e-5)
+
+    def test_block_diagonal_3x4(self):
```
Comment (wasade) [inline gneiss/tests/test_util.py]: what happens if a `nblocks == 1`?

File: gneiss/util.py (hunk 1 above)
Comment (mortonjt) [inline gneiss/util.py]: Fixing

File: gneiss/tests/test_util.py (hunk 2 above)
Comment (mortonjt) [inline gneiss/tests/test_util.py]: It should throw an error, because that is outside the scope of this function

**Comment by mortonjt** [comment]
@wasade comments addressed.  Thanks for reviewing!

**Comment by mortonjt** [comment]
@wasade can you double check this?
